### PR TITLE
solution: 2302-count-subarrays-with-score-less-than-K added

### PR DESCRIPTION
**Pull request by saloni33**
## Change Summary

Provide summary of changes with issue number if any.

## Checklist

**If you haven't fulfilled the below requirements, your PR won't be reviewed.**

### General 

- [x] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [x] Make sure you've followed the template format
- [x] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [x] Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
- [x] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`)
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
```

</TabItem>

</Tabs>
````
- [x] Make sure you've explained your code well (write comments above each line - not inline)
- [x] Make sure you've written your explanation well and it is easy to understand for beginners
- [x] Make sure you've included Time & Space Complexity Analysis
- [x] Make sure you've included a meaningful approach name for your solution. Don't leave it blank. e.g. `## Approach 1: Two Pointers`.

### For Tutorial Articles

- [ ] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [ ] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).

### CODE REVIEW

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+## Problem Link
+
+https://leetcode.com/problems/count-subarrays-with-score-less-than-k/
+
+## Problem Statement
+
+The $score$ of an array is defined as the $product$ of its sum and its length.
+
+For example, the score of [1, 2, 3, 4, 5] is (1 + 2 + 3 + 4 + 5) * 5 = 75.
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:17]: Use LaTeX for numbers - $[1, 2, 3, 4, 5]$

Use bulletin for this line as per LC

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Given a positive integer array nums and an integer k, return the number of non-empty subarrays of nums whose score is strictly less than k.
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:18]: Follow the same format as per LC problem statement

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+A subarray is a contiguous sequence of elements within an array.
+
+
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:22]: Remove an extra line space here

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+Output: 5
+Explanation: Every subarray except [1,1,1] has a score less than 5.
+[1,1,1] has a score (1 + 1 + 1) * 3 = 9, which is greater than 5.
+Thus, there are 5 subarrays having scores less than 5.
+```
+
+
+**Constraints:**
+
+* $1 <= nums.length <= 105$
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:53]: It's $10^5$, and correct it below constraints as well

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+* $1 <= nums[i] <= 105$
+* $1 <= k <= 1015$
+
+
+## Approach : Sliding Window
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:58]: Approach 1: Sliding Window

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+In this approach we will maintain a sliding window from $nums[i]$ to $nums[j]$, subarray starting with i and ending at j which has score less than $k$.
+sum contains the current sum of element between the window.  
+
+We will start iterating j from 0 to nums.size() - 1, first we will add nums[j] to sum.  
+
+The current sum is denoted by $sum$ and length is $j - i + 1$,
+If the score $sum * (j - i + 1) >= k$, the window is too big, we will remove $nums[i]$ and update $i++$.
+We continue doing this until the score is less than k.  
+
+If we find a subarray nums[i] to nums[j] which has score less than k, we will update answer $ans += j - i + 1$ as there will be j - i + 1 total subarrays in total.
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:69]: Apply LaTeX for nums[i] and wherever you have mentioned literals

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+```
+example [1,2,3,4,5] and let i = 0, j = 2
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:72]: Why separate example here ? If it's related to your explanation then add it there and explain with your example

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        while(j < n) {
+            sum += nums[j];
+           
+            // increment j if score is less than k
+            if(sum * (j - i + 1) < k) {
+                ans += j - i + 1;
+                j++;
+            }
+            
+            else {
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:103]: else not align with code format. 
Please format it correctly

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+[1,2,3,4,5] , [1,2,3,4] , and [1,2,3]
+```
+
+
+Time Complexity: $O(n)$, where $n$ - size of array
+
+Space complexity: $O(1)$ 
+
+<Tabs>
+<TabItem value="cpp" label="c++">
```
Comment (vigneshshiv) [inline solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:83]: Correct this.. <TabItem value="c++" label="C++">

**Review by vigneshshiv — CHANGES_REQUESTED** [review]

**Comment by saloni33** [comment]
@vigneshshiv  done the required changes

File: solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md (hunk 8 above)
Comment (vigneshshiv) [reply solutions/2300-2399/2302-count-subarrays-with-score-less-than-k.md:96]: Why else is hanging separately.  Change it like below

```
If (condition) {
  // logic here
} else {
  // logic here
}
```

**Review by vigneshshiv — CHANGES_REQUESTED** [review]
Looks good, last round changes

**Comment by saloni33** [comment]
@vigneshshiv done. Kindly check

**Review by vigneshshiv — APPROVED** [review]
LGTM! @wingkwong please proceed with final review.

**Pull request merged**